[81X] Offline DQM modules for L1Trigger (stage2 calo layer2)

kreczko opened the pull request:
Backport of #16704.

cmsbuild commented:
A new Pull Request was created by  @kreczko (Luke Kreczko) for CMSSW_8_1_X.

It involves the following packages:

DQMOffline/Configuration
DQMOffline/L1Trigger


@dmitrijus, @cmsbuild, @rekovic, @vanbesien, @mulhearn, @davidlange6 can you please review it and eventually sign? Thanks.
@threus, @rociovilar this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

jimbrooke committed:
use correct TCDS counter for MP7 fat event identification

jimbrooke committed:
check consistency between FED ID and useTCDSEventNumber at construction time

jimbrooke committed:
make variable non const

lgray committed:
add FTL RecHit class

lgray committed:
skeleton of RecHit producers

fwyzard committed:
rework use of TCDS in HLTL1NumberFilter

lgray committed:
skeleton of producers

lgray committed:
skeleton of concrete algos

lgray committed:
add python and concrete algos

lgray committed:
add configuration data

lgray committed:
add timing layer customization for local reco

lgray committed:
fix python errors

lgray committed:
fix up issues from running reco step

lgray committed:
add timing layer to fireworks

lgray committed:
put FastTimeGeometry into CaloGeometry

rovere committed:
Exclude outer material for recoMaterial tuning

The material outside the last sensitive detector is
discarded while retuning the recoMaterial file. Its
inclusion caused problems to the fbrem quantities for
electrons. Until we reach a better understanding on this
topic, we use the old logic of ignoring everything after the
last sensitive tracker detectors. All material inside,
instead, is still assigned to the first sensitive layer
(e.g. the beampipe).

rovere committed:
Improve TrackingRecoMaterialAnalyzer

Few options have been added to the package:

* make the destination folder a configurable

* use always the beamspot to select pointing tracks

* optionally filter selected tracks with respect to a valid
  primary vertex, if available, otherwise fall back to the
  beamspot case.

* Write minimal documentation explaining what each histogram
  is supposed to monitor.

rovere committed:
Adapt to the new directory structure

rovere committed:
Fix Run2 workflow

rovere committed:
Increase number of threads

lgray committed:
add to fireworks geometry dump

lgray committed:
add FTlRecHit drawing

lgray committed:
fix up conversion from mm to cm for reco geometry

rovere committed:
Run materialDumperAnalyzer via DQMOffline

The analyzer materialDumperAnalyzer has been added to the
central sequence DQMOfflineTracking, so that it is regularly
run centrally via the DQM sequence, also during PromptReco.

rovere committed:
Minor speedup

lgray committed:
address code review

rekovic commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 15ab1a6832f5704fad1b29782694b66ec026bb89

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16707/17089/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
5.1 step2
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step2_TTbar+TTbarFS+HARVESTFS.log

</pre>1000.0 step4
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step4_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1306.0 step4
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step4_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>1003.0 step3
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step3_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

</pre>1330.0 step4
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step4_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>135.4 step2
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step2_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>10021.0 step5
<pre>runTheMatrix-results/10021.0_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step5_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>9.0 step4
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step4_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>20034.0 step4
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D1_GenSimHLBeamSpotFull14+DigiFull_2023D1+RecoFullGlobal_2023D1+HARVESTFullGlobal_2023D1/step4_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D1_GenSimHLBeamSpotFull14+DigiFull_2023D1+RecoFullGlobal_2023D1+HARVESTFullGlobal_2023D1.log

</pre>25.0 step4
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step4_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>10024.0 step5
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step5_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>50202.0 step4
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step4_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>136.731 step4
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step4_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>25202.0 step4
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step4_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>21234.0 step4
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D4_GenSimHLBeamSpotFull14+DigiFull_2023D4+RecoFullGlobal_2023D4+HARVESTFullGlobal_2023D4/step4_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D4_GenSimHLBeamSpotFull14+DigiFull_2023D4+RecoFullGlobal_2023D4+HARVESTFullGlobal_2023D4.log

</pre>23234.0 step4
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D5_GenSimHLBeamSpotFull14+DigiFull_2023D5+RecoFullGlobal_2023D5+HARVESTFullGlobal_2023D5/step4_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D5_GenSimHLBeamSpotFull14+DigiFull_2023D5+RecoFullGlobal_2023D5+HARVESTFullGlobal_2023D5.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

pietverwilligen committed:
analyze layers and rolls

pietverwilligen committed:
build layers and change RollDetId to have roll=1

gartung committed:
add override keyword where needed

gartung committed:
EventFilter-EcalRawToDigi: fix clang warning hides overloaded virtual function
by adding using directive and adding parameters to function calls to
disambiguate from base class function call.

cmsbuild committed:
Merge pull request #17080 from gartung/EventFilter-EcalRawToDigi-fix-clang-warning

EventFilter-EcalRawToDigi: fix clang warning hides overloaded virtual function

cmsbuild committed:
Merge pull request #17056 from VinInn/removenonsenseerroralongtrajectory

remove nonsense error along trajectory

davidlange6 committed:
Merge pull request #17018 from gartung/DPGAnalysis-SiStripTools-fix-clang-warnings

DPGAnalysis/SiStripTool: fix clang warnings hides overloaded virtual

davidlange6 committed:
Merge pull request #17020 from gartung/PhysicsTools-JetMCAlgos-fix-clang-warnings

PhysicsTools/JetMCAlgos: fix clang warning hides overloaded virtual

ianna committed:
Add 2023 geometry scenario versions

ebrondol committed:
updating README for flat+phase2 px

ebrondol committed:
updating GeometricDetBuilder for flat&tilted OT

felicepantaleo committed:
checking fourth layer equality with unique id

felicepantaleo committed:
disabling caOnlyOneLastHitPerLayerFilter default

rekovic committed:
Now also unpack triggerCount from TCDS.

lgray committed:
further code review

rekovic committed:
Use unpacked TCDS triggerCount value, as opposed to unpacking TCDS again here.

gartung committed:
Add override keyword where needed.

pietverwilligen committed:
implement real 1 eta partition in fully functional structure

pietverwilligen committed:
implement real 1 eta partition in fully functional structure

pietverwilligen committed:
implement real 1 eta partition in fully functional structure

pietverwilligen committed:
implement real 1 eta partition in fully functional structure

pietverwilligen committed:
implement real 1 eta partition in fully functional structure

pietverwilligen committed:
implement real 1 eta partition in fully functional structure

lgray committed:
code review

lgray committed:
another factor 5 in boosted tau seeds

rekovic committed:
Use getByToken for triggerCount, make it fully configurable, and use trackable parameters.

pietverwilligen committed:
no changes wrt original

cmsbuild committed:
Merge pull request #17078 from calabria/ME0BkgFix90X

Fix ME0 neutron background normalization

cmsbuild committed:
Merge pull request #17040 from lgray/topic_speed_tauregion

Speed improvements for TauJetRegionProducer

cmsbuild committed:
Merge pull request #17083 from felicepantaleo/lastHitQuadrupletChi2Filter

CA - Last hit quadruplet chi2 filter

cmsbuild committed:
Merge pull request #17054 from kpedro88/Phase1-HE13

SiPM pulse code cleanup

davidlange6 committed:
Merge pull request #17017 from gartung/PhysicsTools-IsolationUtils-fix-clang-warnings

PhysicsTools/IsolationUtils: fix clang warning hides overloaded virtual function

davidlange6 committed:
Merge pull request #17027 from gartung/HLTriggerOffline-Exotica-fix-clang-warnings

HLTriggerOffline/Exotica: fix clang warnings

davidlange6 committed:
Merge pull request #17051 from gartung/CondFormat-EcalObjects-test-fix-clang-warnings

CondFormat/EcalObjects/test: fix clang warning in test class

cmsbuild committed:
Merge pull request #16976 from gartung/RecoTracker-DebugTools-fix-clang-warnings

RecoTracker/DebugTools: bug fix beginRun not called because of incorrect parameters

davidlange6 committed:
Merge pull request #16916 from gartung/CondTool-DT-fix-clang-warning-fabs-int

CondTool/DT fix clang warning  -Wabsolute-value

mariadalfonso committed:
M3: remore hadhoc 0.95 scaling for the siPM left over from the HPD

gartung committed:
add override; remove empty functions that do not override base class function

makortel committed:
Explicitly set CanExtend to False

pietverwilligen committed:
give number to the rolls in D1

Commit:
Correct the script and also the IsoTrackFilter cutoffs

cmsbuild committed:
Merge pull request #17087 from lgray/topic_more_tauspeed

Further performance improvement in BoostedTauSeeds

cmsbuild committed:
Merge pull request #16997 from lgray/hgcal_cluster_speed

Speedups to HGCal clustering

cmsbuild committed:
Merge pull request #17092 from bsunanda/Run2-alca73

bsunanda:Run2-alca73 Correct the script and also the IsoTrackFilter cutoffs

mariadalfonso committed:
M3: fix 3ns elay for siPM

rekovic committed:
Fix to unpack triggerCount as int64_t instead of int. FED1024 saves it as uint64. Consume it as int64_t.

lgray committed:
clean up dataformats

Dr15Jones committed:
Improve unrunnable schedule detection

Dr15Jones committed:
Properly clear from the stack unneeded edges

mariadalfonso committed:
update the CMSSW config

mariadalfonso committed:
fix PR number

Martin-Grunewald committed:
CMSHLT-1153event content

Martin-Grunewald committed:
Removal of v4 frozen menu

mariadalfonso committed:
restore wrong chi2 setting

davidlange6 committed:
Merge pull request #17082 from cms-l1t-offline/fix_L1_fat_event_filters_90x_bis

Fix l1 fat event filters 90x bis - now use TCDS unpacked data for L1T

cmsbuild committed:
Merge pull request #17057 from makortel/fixMVA

Fix lostmidfrac in track MVA selection

cmsbuild committed:
Merge pull request #17037 from mariadalfonso/fixADCnoiseM2

HBHE: use proper quantization noise in the M2

mariadalfonso committed:
resynch online and offline parameters

mariadalfonso committed:
store the M3 always in eaux

Dr15Jones committed:
Catch an additional type of unrunnable schedule

Handle case where a module on two different paths where on one
path, a needed data product is created earlier while on another
path it is not and the data product can be obtained from an EndPath.

cmsbuild committed:
Merge pull request #17042 from fwyzard/eras_for_HLT_90x_alt

 implement Eras for HLT (90x) (alternative approach)

Dr15Jones committed:
Avoid checking a cycle that already passed

If a crossing edge just has the exact same cycle as a fundamental
cycle, there is no need to recheck the cycle.

mariadalfonso committed:
resynch online and offline parameters

Dr15Jones committed:
Removed unneeded code from cycle detection

Removed parts of the algorithm which had no affect on the results.

Dr15Jones committed:
Moved mix module to prevalidation path

The mix module in the validation end path created the possibility
of an unrunnable schedule because of a data/sequence cycle.

Dr15Jones committed:
Removed commented out debugging code

davidlange6 committed:
Merge pull request #17102 from Dr15Jones/cycleFixWorkInProgress

Better detection of unrunnable scheduled

cmsbuild committed:
Merge pull request #16849 from kskovpen/CMSSW_9_0_X_SimFilter

Muon SIM filter

cmsbuild committed:
Merge pull request #16979 from ikrav/egm_id_80X_v2_rebased_CMSSW_9_0_X_2016-12-07-2300

EGM-certified electron IDs for 2016 data: cut-based and MVA

Commit:
Correct the validation sequence for HGCal

davidlange6 committed:
Merge pull request #17067 from ahinzmann/FixPuppiJets90X

Fix PUPPI lowPUCorr config 90X

davidlange6 committed:
Merge pull request #16901 from lgray/topic_ftl_rechits

Fast timing layer RecHits and event display

mmusich committed:
add new style of e/gamma regressions labels in Run1 MC Global Tags

lgray committed:
get rid of compiler warnings in conversion tools and EcalUncalibRecHitRecAnalFitAlgo

lgray committed:
remove compiler warning in boosted taus

wmtford committed:
Update stripClusterMCanalysis config to 9_0_0_pr2

cmsbuild committed:
Merge pull request #17108 from lgray/topic_remove_warning_boostedtaus

Remove compiler warning in boosted taus

cmsbuild committed:
Merge pull request #17107 from lgray/topic_kill_compiler_warnings

Remove compiler warnings in E/gamma packages

ebrondol committed:
second tuning of PixelShapeCut

Martin-Grunewald committed:
?

Martin-Grunewald committed:
Residual

ronchese committed:
bug fixes:
- BPHDecayVertex
  - check for valid tracks in "validVertex"
  - use clone address as map key in "addV"

- BPHPlusMinusCandidate
  - pass string by reference in "build"

- BPHDecayMomentum
  - add pointer check in "getDaug"

ronchese committed:
bug fixes:

- BPHKx0ToKPiBuilder
  - misleading collection names changed

- BPHOniaToMuMuBuilder
  - retrieve original reco::Candidate only once before loop
    in "getOriginalCandidate"
  - get track search lists from original onia candidates in "extractList"

- TestBPHSpecificDecay
  - fix K name for Bu reconstruction in "analyze"

ronchese committed:
BPHDecayVertex:
new function "getTrackSearchList"

ronchese committed:
BPHTrackReference
- new parameter added to "getTrack" to retrieve the flag for the function
  actually used to get the reco::Track

ronchese committed:
BPHMultiSelect
- selection at kinematic fit level added
- cosmetic: code realigned

ronchese committed:
BPHMultiSelect
- missing function implementation added

ronchese committed:
BPHDecayMomentum
- cascade daughter retrieval added for composite daughters in "getComp"

ronchese committed:
BPHOniaToMuMuBuilder
- selection at kinematic fit level added in "getList"

ronchese committed:
BPHWriteSpecificDecay
- momentum from kinematic fit written on demand in "write"

lgray committed:
initialize timeError properly

Dr15Jones committed:
Fix exception message context information

If an exception occurred, the exception message context would contain
two lines referring to the module on the originating path. This was
from both the Worker and the Path printing module related information.
Now the module info only comes from the Worker.

wmtford committed:
TrackerHitAssociator mods to accommodate Phase 2

Commit:
bringing Phase2 ZS in sync with Phase1

cmsbuild committed:
Merge pull request #17106 from ebrondol/updatingDocTk

Updating Phase2 tracker documentation

cmsbuild committed:
Merge pull request #17096 from cms-tsg-storm/MigrationTo900pre2Template90X

Migration of HLT to 900pre2 template & cleanup

davidlange6 committed:
Merge pull request #17094 from mariadalfonso/M2M3cleanup

HBHE: M3 scale fixes + online/offline M2 parameter adjustements

cmsbuild committed:
Merge pull request #17014 from gartung/EgammaAnalysis-ElectronTools-fix-clang-warnings

EgammaAnalysis/ElectronTools: potential bug fix, add parens to set implied order of + and ?

Sam-Harper committed:
fixing bug where the EcalDriven cut could never be ignored

hroskes committed:
inequality operators for cms types

Dr15Jones committed:
Fix randomly failing framework unit test

The test run_global_stream_one.sh includes a test to check that
when using modules which share a resource that only one module
runs at a time. Previously, the measurement mechanism also included
TriggerResults and the source in the timing which could create
false positives and fail the test. Now the test excludes those
two from the measurement so the test should never see more than one
module running at a time.

Sam-Harper committed:
removing the ostreamstring for the exception messages

wddgit committed:
Prevent module used for cloning from being loaded into process and setup for unscheduled execution

hroskes committed:
tests for numbers and strings

Dr15Jones committed:
Fixed consumes in EcalDetailedTimeRecHitProducer

EcalDetailedTimeRecHitProducer was calling consumes for data products
unconditionally even though there were parameters which caused the
data products to not be gotten. Now the parameters also conditionally
protect the consumes calls to make the code internally consistent.

hroskes committed:
make double("5") invalid

Sam-Harper committed:
initial test class

cmsbuild committed:
Merge pull request #17119 from Dr15Jones/fixUnitTest

Fix randomly failing framework unit test

cmsbuild committed:
Merge pull request #17112 from lgray/topic_fix_ftltimeerror

Initialize timeError properly in FTLUncalibratedRecHit

davidlange6 committed:
Merge pull request #16913 from Sam-Harper/HEEPV70VID_90X

HEEP V7.0 in 90X

cmsbuild committed:
Merge pull request #16921 from gartung/CondFormats-SiPixelObjects-fix-clang-warnings

CondFormats-SiPixelObjects fix clang warnings about abs and mismatched tags

Sam-Harper committed:
adding more types of value map

Sam-Harper committed:
adding in the ability to select the type stored in the PATObject

Commit:
Fix empty list of LHE files

Commit:
Add missing HcalDeterministicFit::rCorrSiPM

The following error in blocking Clang builds:

    RecoLocalCaloHcalRecAlgos/HcalDeterministicFit.o: In function calDeterministicFit::phase1Apply(HBHEChannelInfo const&, float&, float&) const':
    RecoLocalCalo/HcalRecAlgos/src/HcalDeterministicFit.cc:(.text+0x583): undefined reference to calDeterministicFit::rCorrSiPM'
    RecoLocalCalo/HcalRecAlgos/src/HcalDeterministicFit.cc:(.text+0x5a9): undefined reference to calDeterministicFit::rCorrSiPM'

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

hroskes committed:
change cms.double("0.5") to cms.double(0.5)

hroskes committed:
and some more

knoepfel committed:
Improvements to stall grapher.

- Significant improvements to plotting efficiency.
- Fixes to thread accounting--not fully solved, but better.

knoepfel committed:
Emit timing information for preSourceEvent and postSourceEvent signals--update stall-monitoring script.

ahinzmann committed:
get rid of mayConsume

ahinzmann committed:
use existing weights when running on top of MiniAOD to make results from AOD and MiniAOD more similar

cmsbuild committed:
Merge pull request #17126 from davidlt/fix-clang-rCorrSiPM

Add missing HcalDeterministicFit::rCorrSiPM

davidlange6 committed:
Merge pull request #17124 from srimanob/EMPTY_LHE_FIX-90X

Fix empty list of LHE files (90X)

davidlange6 committed:
Merge pull request #17113 from Dr15Jones/fixExceptionContextMessage

Fix exception message context information

cmsbuild committed:
Merge pull request #17118 from Sam-Harper/EcalDrivenCutFix90X

bug fix for GsfEleEcalDrivenCut

davidlange6 committed:
Merge pull request #17121 from wddgit/from-CMSSW_9_0_X_2017-01-05-1100

Prevent module used for cloning from being loaded into process and se…

cmsbuild committed:
Merge pull request #17120 from Dr15Jones/fixConsumesEcalDetailedTimeRecHitProducer

Fixed consumes in EcalDetailedTimeRecHitProducer

davidlange6 committed:
Merge pull request #17081 from ianna/dump-sim-geom-2023-versioned

Add 2023 Geometry Scenario Version

Sam-Harper committed:
reordering file location and cleaning up obsolete classes

davidlange6 committed:
Merge pull request #16983 from gartung/CalibCalorimetry-HcalAlgos-test-fix-clang-warnings

CalibCalorimetry/HcalAlgos/test fix clang warning hides overloaded virtual function

cmsbuild committed:
Merge pull request #17104 from cms-AlCaDB/from-CMSSW_9_0_X_updateRun1MC_GTs_forEgamma

[90X] Add new style of e/gamma regressions labels in Run1 MC Global Tags

Commit:
minimal update HE only needed

slehti committed:
Changed hltCaloStage2Digis -> caloStage2Digis

smuzaffar committed:
filter das_client results match the needed lumi sections if requested

lgray committed:
remove use of tmpnam

Dr15Jones committed:
Force untracked parameters for modules into Registry

If an input file contained the same PSet ID (because of identical
tracked parameters) as a module in a process, the untracked
parameters for the module are not available in the Registry. This
caused problems with the deleteEarly system which uses an untracked
parameter. We now force the PSet for a module to be injected into
the Registry.

cmsbuild committed:
Merge pull request #17137 from Dr15Jones/rememberUntracked

Force untracked parameters for modules into Registry

cmsbuild committed:
Merge pull request #17135 from lgray/topic_fix_tmpnamwarning

Remove use of tmpnam in generators

cmsbuild committed:
Merge pull request #17114 from abdoulline/HCAL_Phase2_ZS_fix

Bringing Phase2 ZS in sync with Phase1 one

cmsbuild committed:
Merge pull request #17103 from bsunanda/Phase2-hgx72

bsunanda:Phase2-hgx72 Correct the validation sequence for HGCal

cmsbuild committed:
Merge pull request #17091 from makortel/fixTrackingPlotting

Fix crash in tracking MC validation plot scripts

cmsbuild committed:
Merge pull request #17076 from mtosi/fixHLTonlineDQM

add Throughput client

cmsbuild committed:
Merge pull request #16973 from schneiml/onlinedqm-fileinput

Make the online DQM fileinputsource more intelligent

Commit:
FriendlyName: add "String" options for C++11 ABI

This is needed for C++11 (and above) ABI, which is available starting GCC 5.

Signed-off-by: David Abdurachmanov <davidlt@cern.ch>

smuzaffar committed:
For runTheMatrix: fix lumi section selection logic to avoid duplicate file names

knoepfel committed:
Plotting/script improvements.

knoepfel committed:
Fix scripting errors.

davidlange6 committed:
Merge pull request #17140 from cms-sw/backport-CMSSW_9_0_X-17133

filter das_client results match the needed lumi sections if requested

cmsbuild committed:
Merge pull request #17139 from davidlt/friendly_name_add_string_cxx11_abi

FriendlyName: add "String" options for C++11 ABI

cmsbuild committed:
Merge pull request #17109 from wmtford/clusNtp900pr2

Update stripClusterMCanalysis config to 9_0_0_pr2

cmsbuild committed:
Merge pull request #16749 from ghellwig/geometry-independent-PCL_90X

[90X] Geometry independent SiPixelAli PCL

cmsbuild committed:
Merge pull request #17066 from cms-AlCaDB/90X_upgrade2023_geometries

[90X] update 2023 upgrade simulation Global Tag with SIM and RECO geometries tags

smuzaffar committed:
Merge pull request #17142 from cms-sw/backport-CMSSW_9_0_X-17141

For runTheMatrix: fix lumi section selection logic to avoid duplicate file names

ronchese committed:
cosmetics in BPHWriteSpecificDecay

archiron committed:
rescaling of provenance histos updated

archiron committed:
no modification

archiron committed:
no modification

archiron committed:
no modification

Dr15Jones committed:
Fix L1TCaloParamsCompare operator==

The operator== function only compared the first object to itself
rather than comparing the first object to the second object.
Found by gcc 6.2.

Commit:
Fix self-comparison in L1MuGMTExtendedCand

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

wddgit committed:
Replace import * when modules are imported only to be cloned

Dr15Jones committed:
Fix operator!= for pat::strbitset

The operator!= was calling itself to form an infinite recursion.
It now calls ! operator== which was the original intent.

gartung committed:
Fix error from apple-clang-8.0 because of missing #include <functional>

While compiling with apple-clang-8.0 many error were the result of a missing include
in DataFormats/Provenenace/interface/Hash.h. The gcc header <string> includes
<functional> but the apple clang's version does not.

This is the error.

/private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-_shPan/spack-build/CMSSW_9_0_0_pre2/src/DataFormats/Provenance/interface/Hash.h:165:43: error: no m
ember named 'greater' in namespace 'std'
    return this->compareUsing(other, std::greater<std::string>());
                                     ~~~~~^
/private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-_shPan/spack-build/CMSSW_9_0_0_pre2/src/DataFormats/Provenance/interface/Hash.h:165:62: error: expe
cted '(' for function-style cast or type construction
    return this->compareUsing(other, std::greater<std::string>());
                                                  ~~~~~~~~~~~^
/private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-_shPan/spack-build/CMSSW_9_0_0_pre2/src/DataFormats/Provenance/interface/Hash.h:165:64: error: expe
cted expression
    return this->compareUsing(other, std::greater<std::string>());
                                                               ^
/private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-_shPan/spack-build/CMSSW_9_0_0_pre2/src/DataFormats/Provenance/interface/Hash.h:172:43: error: no t
emplate named 'equal_to' in namespace 'std'; did you mean '__equal_to'?
    return this->compareUsing(other, std::equal_to<std::string>());
                                     ~~~~~^~~~~~~~
                                          __equal_to
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:653:8: note: '__equal_to' declared here
struct __equal_to
       ^

wddgit committed:
Remove import * for cloned modules

cmsbuild committed:
Merge pull request #17155 from gartung/DataFormats-Provenance-fix-apple-clang-error

Fix error from apple-clang-8.0 because of missing #include <functional>

cmsbuild committed:
Merge pull request #17146 from archiron/Renorm_Provenance_Histos_90X_V1

Provenance histos rescaling updated 90X v1

davidlange6 committed:
Merge pull request #17111 from ronchese/bph_reco_170104

Bph reco update, jan 01 2017

davidlange6 committed:
Merge pull request #16999 from bsunanda/Phase2-hgx71

bsunanda:Phase2-hgx71 Update all validation code in view of BH changes

wmtford committed:
Print out more info in TestAssociator

davidlange6 committed:
Merge pull request #17117 from hroskes/inequality-operators-90X

inequality operators for cms types

davidlange6 committed:
Merge pull request #17072 from ahinzmann/PuppiPhotonFix90X

Fix PUPPI MET tails

davidlange6 committed:
Merge pull request #17152 from Dr15Jones/fixNotEqualOp

Fix operator!= for pat::strbitset

Commit:
Fix warning (non-null compare) in MatrixMeschach

This is not legal according to C++, i.e., this should never be
0x0/NULL/nullptr.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Fix FORTRAN warnings in ME2pythia.f

This seems to be written in FORTRAN 77 (decades old standard). These
warnings:

    Warning: Legacy Extension: Comma before i/o item list at (1)

to GNU FORTRAN extension "6.1.7 I/O item lists":

    To support legacy codes, GNU Fortran allows the input item list of
    the READ statement, and the output item lists of the WRITE and PRINT
    statements, to start with a comma.

The patch removes the commas.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

smuzaffar committed:
use cmssw_das_client wrapper instead of duplicated das_client code

wddgit committed:
Use _m instead of m as a variable name.

pietverwilligen committed:
new ME0 GeometryBuilder for D4

pietverwilligen committed:
new ME0 GeometryBuilder for D4

pietverwilligen committed:
new ME0 GeometryBuilder for D7

pietverwilligen committed:
new ME0 GeometryBuilder for D8

pietverwilligen committed:
new ME0 GeometryBuilder for D8

pietverwilligen committed:
new ME0 GeometryBuilder for D10

pietverwilligen committed:
new ME0 GeometryBuilder for M1

gartung committed:
Fixes for changes in location of apple clang's libc++ header locations. These are ifdef with __APPLE__ so they should not interfere with linux compilations. I had to add -I/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/ to ROOT_ICLUDE_PATH before /usr/include/c++/4.2.1 to ensure that the correct header were picked up.

gartung committed:
__gnu__cxx is not define for apple clang

gartung committed:
don't need to use these anymore

davidlange6 committed:
Merge pull request #17167 from smuzaffar/use-cmssw-das-in-edmPickEvent

use cmssw_das_client wrapper instead of duplicated das_client code

cmsbuild committed:
Merge pull request #17166 from davidlt/fix-fortran77-comma

Fix FORTRAN warnings in ME2pythia.f

cmsbuild committed:
Merge pull request #17164 from davidlt/fix-non-null-comp-MatrixMeschach

Fix warning (non-null compare) in MatrixMeschach

davidlange6 committed:
Merge pull request #17158 from wddgit/JME

Remove import * for cloned modules

cmsbuild committed:
Merge pull request #17151 from davidlt/fix-self-comp-L1MuGMTExtendedCand

Fix self-comparison in L1MuGMTExtendedCand

cmsbuild committed:
Merge pull request #17150 from Dr15Jones/gcc6WarningFix

Fix L1TCaloParamsCompare operator==

davidlange6 committed:
Merge pull request #17138 from Sam-Harper/NewEGModifiers_8010

New E/gamma pat::Electron and pat::Photon modifiers

davidlange6 committed:
Merge pull request #17153 from wddgit/from-CMSSW_9_0_X_2017-01-11-1100

Replace import * when modules are imported only to be cloned

Dr15Jones committed:
Hide debugging code in TrackingTrughAccumulator

gcc 6.3 was giving a warning about unnused debugging code. Using
a preprocessor switch allows the code to be skipped for normal
compilation but kept for debugging.

rovere committed:
Move RecoMaterialAnalyzer in DQM

Dr15Jones committed:
Use dynamic_cast instead of reinterpret_cast in HeavyFlavorAnalysis

The original code assumed calling reinterpret_cast on a pointer
would return 0 if the pointer was of the wrong type. That is not
the case. The code was changed to use dynamic_cast instead which
does have that behavior.

Dr15Jones committed:
Replaced tab with space to avoid gcc 6.3 warning in L1Trigger/L1TCommon

Because of the different possible spacing used for tabs, gcc 6.3
was warning of a line after an `if` that might appear to visually
be part of the `if` but is, in fact, not. Switching from tabs to
using spaces, which is consistent with the rest of the document,
removes the warning.

pietverwilligen committed:
cleanup

pietverwilligen committed:
simplify by changing default false -> true

pietverwilligen committed:
simplify by changing default false -> true

pietverwilligen committed:
simplify by changing default false -> true

pietverwilligen committed:
simplify by changing default false -> true

pietverwilligen committed:
simplify by changing default false -> true

pietverwilligen committed:
simplify by changing default false -> true

pietverwilligen committed:
simplify by changing default false -> true

knoepfel committed:
Fix boundary-conditions error.

lgray committed:
add timing to GeomDetEnumerators

davidlange6 committed:
Merge pull request #17174 from Dr15Jones/fixSpacing

Replaced tab with space to avoid gcc 6.3 warning in L1Trigger/L1TCommon

davidlange6 committed:
Merge pull request #17173 from Dr15Jones/useDynamicCast

Use dynamic_cast instead of reinterpret_cast in HeavyFlavorAnalysis

cmsbuild committed:
Merge pull request #17172 from Dr15Jones/hideDebuggingCode

Hide debugging code in TrackingTrughAccumulator

cmsbuild committed:
Merge pull request #17168 from gartung/DataFormats-N-FWCore-apple-clang-fixes

Fixes for changes in location of apple clang's libc++ header locations

cmsbuild committed:
Merge pull request #17165 from slehti/HLTTauValidationUpdate_CMSSW90X

Changed hltCaloStage2Digis -> caloStage2Digis

cmsbuild committed:
Merge pull request #17127 from knoepfel/stall-grapher

Improvements to stall-monitoring service and script.

cmsbuild committed:
Merge pull request #17110 from ebrondol/phase2FixClusShapeFilter_90X

Second tuning of Pixel cluster shape cut for Phase 2

cmsbuild committed:
Merge pull request #17176 from lgray/topic_add_timing_GeomDetEnumerators

Add timing detectors to GeomDetEnumerators

cmsbuild committed:
Merge pull request #17175 from pietverwilligen/90X_NewME0GeometryBuilder_ForPR_v1

New ME0GeometryBuilder for baseline ME0 geometry

cmsbuild committed:
Merge pull request #17161 from wmtford/hitAssocForPhase2

Hit assoc for phase2

cmsbuild committed:
Merge pull request #17134 from rovere/AddRecoMaterialToDQM

Add reco material to dqm

cmsbuild commented:
Pull request #16707 was updated. @dmitrijus, @cmsbuild, @rekovic, @vanbesien, @mulhearn, @davidlange6 can you please check and sign again.

kreczko committed:
removed obsolete file

kreczko committed:
added a more general L1T Efficiency Harvesting

kreczko committed:
added "diff" module for emulator comparison

kreczko committed:
containing under- and over-flow bins within histogram limits

kreczko committed:
added L1TStage2CaloLayer2Offline for offline DQM

cmsbuild commented:
Pull request #16707 was updated. @ghellwig, @ianna, @cerminar, @fabozzi, @Martin-Grunewald, @rekovic, @govoni, @perrotta, @civanch, @silviodonato, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @Dr15Jones, @cvuosalo, @mdhildreth, @slava77, @ggovi, @mmusich, @mulhearn, @arunhep, @perrozzi, @thuer, @monttj, @srimanob, @hengne can you please check and sign again.

slava77 commented:
-1

looks like this was submitted to a wrong branch (81X instead of 90X), or something else is similarly wrong.

kreczko commented:
@slava77 Yes, superseded by  #17189:

> The initial version of #16707 used the same branch as the 90X PR (#16704).
> However, since now 90X and 81X grew apart, a separate branch is needed for the backport.
